beacon_node/beacon_chain/src/blob_verification.rs

```diff
@@ -147,63 +190,169 @@ pub type GossipVerifiedBlobList<T> = VariableList<
 /// the p2p network.
 #[derive(Debug)]
 pub struct GossipVerifiedBlob<T: BeaconChainTypes> {
-    blob: SignedBlobSidecar<T::EthSpec>,
+    block_root: Hash256,
+    blob: KzgVerifiedBlob<T::EthSpec>,
+}
+
+impl<T: BeaconChainTypes> Deref for GossipVerifiedBlob<T> {
+    type Target = BlobSidecar<T::EthSpec>;
```

Comment:
The [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/predictability.html) recommend that only smart-pointer types should implement `Deref` and `DerefMut`. Further [my Rust book](https://www.oreilly.com/library/view/programming-rust-2nd/9781492052586/) notes:

> You should not implement `Deref` and `DerefMut` for a type just to make the `Target` type's methods appear on it automatically, the way a C++ base class's methods are visible on a subclass. This will not always work as you expect and can be confusing when it goes awry.

Reply:
Good call